packages/amplify-graphql-api-construct/src/__tests__/internal/data-source-config.test.ts

```diff
@@ -1,75 +1,124 @@
 
 describe('datasource config', () => {
-  it('should parse the datasource config correctly', () => {
-    const input: Record<string, ModelDataSourceStrategy> = {
-      Todo: {
-        dbType: 'DYNAMODB',
-        provisionStrategy: 'DEFAULT',
-      },
-      Author: DDB_AMPLIFY_MANAGED_DATASOURCE_STRATEGY,
-      Post: {
-        name: 'mysqlTable',
-        dbType: 'MYSQL',
-        dbConnectionConfig: {
-          hostnameSsmPath: 'hostnameSsmPath',
-          portSsmPath: 'portSsmPath',
-          usernameSsmPath: 'usernameSsmPath',
-          passwordSsmPath: 'passwordSsmPath',
-          databaseNameSsmPath: 'databaseNameSsmPath',
+  const dbConnectionConfig: SqlModelDataSourceDbConnectionConfig = {
+    hostnameSsmPath: '/test/hostname',
+    portSsmPath: '/test/port',
+    usernameSsmPath: '/test/username',
+    passwordSsmPath: '/test/password',
+    databaseNameSsmPath: '/test/databaseName',
+  };
+
+  it('constructCustomSqlDataSourceStrategies', () => {
+    const schema = /* GraphQL */ `
+      type Todo @model {
+        id: ID! @primaryKey
+        description: String
+      }
+      type Query {
+        myCustomQuery: [Int] @sql(statement: "SELECT 1")
+      }
+      type Mutation {
+        myCustomMutation: [Int] @sql(statement: "UPDATE mytable SET id=1; return 1;")
+      }
+    `;
+
+    const mysqlStrategy: SQLLambdaModelDataSourceStrategy = {
+      name: 'mysqlStrategy',
+      dbType: 'MYSQL',
+      dbConnectionConfig,
+    };
+
+    const sqlDataSourceStrategies = constructCustomSqlDataSourceStrategies(schema, mysqlStrategy);
+    expect(sqlDataSourceStrategies).toEqual(
+      expect.arrayContaining([
+        {
+          typeName: 'Query',
+          fieldName: 'myCustomQuery',
+          strategy: mysqlStrategy,
         },
-      },
-      Comment: {
-        name: 'pgTable',
-        dbType: 'POSTGRES',
-        dbConnectionConfig: {
-          hostnameSsmPath: 'hostnameSsmPath',
-          portSsmPath: 'portSsmPath',
-          usernameSsmPath: 'usernameSsmPath',
-          passwordSsmPath: 'passwordSsmPath',
-          databaseNameSsmPath: 'databaseNameSsmPath',
+        {
+          typeName: 'Mutation',
+          fieldName: 'myCustomMutation',
+          strategy: mysqlStrategy,
         },
+      ]),
+    );
+  });
+
+  it('getDataSourceStrategiesProvider for a single definition', () => {
+    const schema = /* GraphQL */ `
+      type Todo @model {
+        id: ID! @primaryKey
+        description: String
+      }
+      type Query {
+        myCustomQuery: [Int] @sql(reference: "myCustomQuery")
+      }
+      type Mutation {
+        myCustomMutation: [Int] @sql(reference: "myCustomMutation")
+      }
+    `;
+
+    const strategy: SQLLambdaModelDataSourceStrategy = {
+      name: 'strategy',
+      dbType: 'MYSQL',
+      dbConnectionConfig,
+      customSqlStatements: {
+        myCustomQuery: 'SELECT 1;',
+        myCustomMutation: 'UPDATE mytable SET id=1; return 1;',
```

Comment:
If I'm not wrong, `RETURN` clause can be used with functions only. Can we replace it with `SELECT 1` or any valid statement?